src/plugins/InlineSpoilers/Plugin.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  Mark,
+  markInputRule,
+  markPasteRule,
+  mergeAttributes,
+} from "@tiptap/core";
+
+import { PluginKey } from "@tiptap/pm/state";
+import { toggleAttributeOnClick } from "../utils";
+
+export interface InlineSpoilersOptions {
+  visible?: boolean;
+  focusable?: boolean;
+}
+
+export const InlineSpoilersPluginKey = new PluginKey("InlineSpoilersPlugin");
+
+export const PLUGIN_NAME = "inlineSpoilers";
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    [PLUGIN_NAME]: {
+      setInlineSpoilers: (attributes?: { visible: boolean }) => ReturnType;
+      toggleInlineSpoilers: (attributes?: { visible: boolean }) => ReturnType;
+      unsetInlineSpoilers: () => ReturnType;
+    };
+  }
+}
+
+export const inputRegex = /(?:^|\s)((?:\|\|)((?:[^\|]+))(?:\|\|))$/;
```

Comment:
Can you add a comment on what this regex is looking for and what is supposed to match/not match?

Reply:
done